src/ChunkData.cpp

```diff
@@ -325,18 +305,54 @@ NIBBLETYPE cChunkData::GetSkyLight(int a_RelX, int a_RelY, int a_RelZ) const
 
 
 
-cChunkData cChunkData::Copy(void) const
+const cChunkData::sChunkSection * cChunkData::GetSection(size_t a_SectionNum) const
 {
-	cChunkData copy(m_Pool);
-	for (size_t i = 0; i < NumSections; i++)
+	if (a_SectionNum < NumSections)
+	{
+		return m_Sections[a_SectionNum];
+	}
+	ASSERT(!"cChunkData::GetSection: section index out of range");
+	return nullptr;
+}
+
+
+
+
+
+UInt16 cChunkData::GetSectionBitmask() const
+{
+	static_assert(NumSections <= 16U, "cChunkData::GetSectionBitmask needs a bigger data type");
+	UInt16 Res = 0U;
+	for (size_t i = 0U; i < NumSections; ++i)
+	{
+		Res |= ((m_Sections[i] != nullptr) << i);
+	}
+	return Res;
+}
+
+
+
+
+
+void cChunkData::Clear()
+{
+	for (size_t i = 0; i < NumSections; ++i)
 	{
 		if (m_Sections[i] != nullptr)
 		{
-			copy.m_Sections[i] = copy.Allocate();
-			*copy.m_Sections[i] = *m_Sections[i];
+			Free(m_Sections[i]);
+			m_Sections[i] = nullptr;
 		}
 	}
-	return copy;
+}
+
+
+
+
+
+cChunkData cChunkData::Copy(void) const
```

Comment:
This should be the using the code from the copy constructor so that copies are explict, as they are really expensive.